src/esp/gfx_batch/Renderer.cpp

```diff
@@ -1076,10 +1076,6 @@ Cr::Containers::StridedArrayView1D<Mn::Float> Renderer::lightRanges(
 }
 
 void Renderer::draw(Mn::GL::AbstractFramebuffer& framebuffer) {
-  // TODO allow this (currently addFile() sets up shader limits)
-  CORRADE_ASSERT(!state_->meshes.isEmpty(),
-                 "Renderer::draw(): no file was added", );
-
```

Comment:
The reason there's an assertion and a TODO is that the shaders aren't initialized if `addFile()` wasn't called, and IIRC that led to a GL error somewhere in the code below.

To pin down the behavior, can you add a dedicated test case for this into `GfxBatchRendererTest.cpp` and verify that it doesn't produce a GL error (with `MAGNUM_VERIFY_NO_GL_ERROR()`)? If it does, then it probably needs an early out in case there's nothing to draw from.

Reply:
Sounds good!